ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ContractKeysIT.scala

```diff
@@ -362,4 +362,31 @@ final class ContractKeysIT extends LedgerTestSuite {
         assertGrpcError(failedFetch, Status.Code.INVALID_ARGUMENT, "not visible")
       }
   })
+
+  test(
+    "CKDivulgedContractKeyReusability",
+    "Subsequent divulged contracts can use the same contract key",
+    allocate(SingleParty, SingleParty),
+  )(implicit ec => {
+    case Participants(Participant(ledger1, party1), Participant(ledger2, party2)) =>
+      for {
+        creator1 <- ledger1.create(party1, WithKeyCreator(party1, party2))
+        withKey1 <- ledger1.exerciseAndGetContract[WithKey](
```

Comment:
I think the other direction is interesting as well where this command is submitted on ledger 2, see https://gist.github.com/cocreature/c134d2ba0dd974708ed061a5ec053256 for an example. They both run into the same error at the moment but I think it’s helpful to test for both.

Reply:
If you want to be extra thorough test the cases where the first gets created on p1 and the second on p2 and the other way around.

Reply:
You're right, it's worth testing the second case. Actually I'm currently working on it - @mziolekda sent me the scenario.